### DISCUSSION TITLE
mkdir during put/tee?

### DISCUSSION THREAD

**Discussion by srstsavage** [johnkerl/miller#1096] [Q&A] [answered]
Hi, does anyone know of a performant way to create directories while using put/tee?

The following completes in about 1 second.
```
mlr --csv put -q 'tee > "test-".$Date.".csv", $*' data.csv
```

But this takes over a minute.

```
mlr --csv put -q 'system("mkdir -p date-hierarchy/".$Date); tee > "date-hierarchy/".$Date."/test-".$Date.".csv", $*' data
```

**Comment by srstsavage** [comment, marked as answer]
Actually, doing directory creation as a separate first step seems to be the way to go. Both of these complete in less than a second:

```
mlr --csv uniq -g Date then put -q 'system("mkdir -p date-hierarchy/".$Date)' data.csv
```

```
mlr --csv --headerless-csv-output uniq -g Date data.csv | xargs -I{} mkdir -p date-hierarchy/{}
```

I am curious if it's possible to do this in a single step with `tee`, but the two-step approach is perfectly acceptable.

**Comment by johnkerl** [comment]
@shane-axiom sounds great to me! :)
